components/feature/recentlyclosed/src/test/java/mozilla/components/feature/recentlyclosed/RecentlyClosedMiddlewareTest.kt

```diff
@@ -109,6 +111,7 @@ class RecentlyClosedMiddlewareTest {
 
             store.dispatch(TabListAction.RemoveTabAction("1234")).joinBlocking()
 
+            store.waitUntilIdle()
             dispatcher.advanceUntilIdle()
```

Comment:
I think we need to flip the order here: Fist we need to wait for the middleware coroutine to complete and dispatch on the store, then, when we know the store will already know about the action, wait for the store.